functions/Copy-DbaResourceGovernor.ps1

```diff
@@ -109,26 +113,93 @@ function Copy-DbaResourceGovernor {
+            Notes             = $null
+            DateTime          = [DbaDateTime](Get-Date)
+        }
 
         if ($Pscmdlet.ShouldProcess($destination, "Updating Resource Governor settings")) {
             if ($destServer.Edition -notmatch 'Enterprise' -and $destServer.Edition -notmatch 'Datacenter' -and $destServer.Edition -notmatch 'Developer') {
                 Write-Message -Level Verbose -Message "The resource governor is not available in this edition of SQL Server. You can manipulate resource governor metadata but you will not be able to apply resource governor configuration. Only Enterprise edition of SQL Server supports resource governor."
             }
             else {
                 try {
-                    $sql = $sourceServer.ResourceGovernor.Script() | Out-String
-                    Write-Message -Level Debug -Message $sql
-                    Write-Message -Level Verbose -Message "Updating Resource Governor settings."
-                    $destServer.Query($sql)
+                    Write-Message -Level Verbose -Message "Managing classifier function."
+                    if (!$sourceClassifierFunction) {
+                        $copyResourceGovClassifierFunc.Status = "Skipped"
+                        $copyResourceGovClassifierFunc.Notes = $null
+                        $copyResourceGovClassifierFunc | Select-DefaultView -Property DateTime, SourceServer, DestinationServer, Name, Type, Status, Notes -TypeName MigrationObject
+                    }
+                    else {
+                        $fullyQualifiedFunctionName = $sourceClassifierFunction.Schema+"."+$sourceClassifierFunction.Name
+                                  
+                        if (!$destClassifierFunction) {
+                            $destFunction = $destServer.Databases["master"].UserDefinedFunctions[$sourceClassifierFunction.Name]
+
+                            if ($destFunction) {
+                                Write-Message -Level Verbose -Message "Dropping the function with the source classifier function name."
+                                $destFunction.Drop()
+                            }
 
-                    $copyResourceGovSetting.Status = "Successful"
-                    $copyResourceGovSetting | Select-DefaultView -Property DateTime, SourceServer, DestinationServer, Name, Type, Status, Notes -TypeName MigrationObject
+                            Write-Message -Level Verbose -Message "Creating function."
+                            $destServer.Query($sourceClassifierFunction.Script())
+                            
+                            $sql = "ALTER RESOURCE GOVERNOR WITH (CLASSIFIER_FUNCTION = $fullyQualifiedFunctionName);"
+                            $sql
+                            Write-Message -Level Debug -Message $sql
+                            Write-Message -Level Verbose -Message "Mapping Resource Governor classifier function."
+                            $destServer.Query($sql)
```

Comment:
The calls to execute a query really should be wrapped in a try/catch block so a proper error can be returned when one of them fails and the user is aware of what area failed.

Reply:
so you're suggesting me to indent the try/catch inside the outer one? I've copied the previous version which did a single block. Would you change fragmenting the commands into different try catch blocks?

Reply:
If any one of those queries fails does the user need to know which one?

Reply:
I think that the IF part and the ELSE part will get just a query to know for in case of error. Queries like `ALTER RESOURCE GOVERNOR WITH (CLASSIFIER_FUNCTION = NULL);`  and `ALTER RESOURCE GOVERNOR RECONFIGURE;` will not fail (error could be connection). Just the final query (in the IF and ELSE parts) needs to be checked.

Reply:
Those queries can fail if permissions do not exist.

Reply:
I'd like to know if I've to change the original code (the one before my changes) splitting in more than an outer try/catch block or indeting two more inside it.